packages/material-ui/src/ObjectFieldTemplate/ObjectFieldTemplate.tsx

```diff
@@ -57,7 +57,7 @@ const ObjectFieldTemplate = ({
           ) : (
             <Grid
               item={true}
-              xs={12}
+              xs={uiSchema['ui:fieldFlexWidth'] || 12}
```

Comment:
As there is more than this one grid in the theme, so I'm not sure your fix is complete. Also, it is missing an update to tests to verify the change. Moreover, in v5 beta, an additional theme for material-ui 5 was added so you might want replicate this there. And finally, other themes also use the flex grid concept, so maybe making the appropriate changes there is a good thing?